MekHQ/src/mekhq/gui/adapter/UnitTableMouseAdapter.java

```diff
@@ -419,17 +419,19 @@ public void actionPerformed(ActionEvent action) {
                 MekHQ.triggerEvent(new UnitChangedEvent(selectedUnit));
             }
         } else if (command.equals(COMMAND_REMOVE_INDI_CAMO)) {
-            for (Unit u : units) {
-                u.getEntity().setCamouflage(new Camouflage());
+            for (final Unit unit : units) {
+                unit.getEntity().setCamouflage(new Camouflage());
             }
-        } else if (command.equals(COMMAND_INDI_CAMO)) { // Single Unit only
-            CamoChooserDialog ccd = new CamoChooserDialog(gui.getFrame(),
+        } else if (command.equals(COMMAND_INDI_CAMO)) {
+            final CamoChooserDialog ccd = new CamoChooserDialog(gui.getFrame(),
                     selectedUnit.getUtilizedCamouflage(gui.getCampaign()), true);
             if (ccd.showDialog().isCancelled()) {
                 return;
             }
-            selectedUnit.getEntity().setCamouflage(ccd.getSelectedItem());
-            MekHQ.triggerEvent(new UnitChangedEvent(selectedUnit));
+            for (final Unit unit : units) {
+                unit.getEntity().setCamouflage(ccd.getSelectedItem());
+                MekHQ.triggerEvent(new UnitChangedEvent(unit));
```

Comment:
So here's a thought: why are we triggering the unit changed event here but not for the remove camo command? Also, why are we triggering it at all?

Reply:
Nice catch... I'll double check if it is required, and either add or remove if it is. IIRC it is supposed to force the icon to update

Reply:
It's a bug with the remove camo command. I've updated it for the future.